cluster-autoscaler/scale_down.go

```diff
@@ -260,6 +271,114 @@ func getEmptyNodes(candidates []*apiv1.Node, pods []*apiv1.Pod, maxEmptyBulkDele
 	return result[:limit]
 }
 
+func deleteNode(node *apiv1.Node, pods []*apiv1.Pod, client kube_client.Interface, cloudProvider cloudprovider.CloudProvider, recorder kube_record.EventRecorder) error {
+	if err := drainNode(node, pods, client, recorder); err != nil {
+		return err
+	}
+	return deleteNodeFromCloudProvider(node, cloudProvider, recorder)
+}
+
+// Performs drain logic on the node. Marks the node as unschedulable and later removes all pods, giving
+// them up to MaxGracefulTerminationTime to finish.
+func drainNode(node *apiv1.Node, pods []*apiv1.Pod, client kube_client.Interface, recorder kube_record.EventRecorder) error {
+	if err := makeUnschedulable(node, client, recorder); err != nil {
+		return err
+	}
+
+	for _, pod := range pods {
+		recorder.Eventf(pod, apiv1.EventTypeNormal, "ScaleDown", "deleting pod for node scale down")
+		err := client.Core().Pods(pod.Namespace).Delete(pod.Name, &apiv1.DeleteOptions{})
```

Comment:
What difference would it make? I mean, we kill the node after MaxGracefulTerminationTime anyway.